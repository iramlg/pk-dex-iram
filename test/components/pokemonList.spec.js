const React = require('react');
const PokemonList = require('../../src/components/pokemonList');
const renderer = require('react-test-renderer');

describe('<PokemonList />', () => {
  describe('render', () => {
    it('test rendering', () => {
      const component = renderer.create(<PokemonList getPokemon={ jest.fn() } list={ [{name: 'name', url: 'url'}] }  />).toJSON();
      expect(component).toMatchSnapshot();
    });

    it('test empty list', () => {
      const component = renderer.create(<PokemonList getPokemon={ jest.fn() } list={ [] }  />).toJSON();
      expect(component).toMatchSnapshot();
    });
  });

  describe('Props', () => {
    it('types', () => {
      expect(Object.keys(PokemonList.propTypes).length).toBe(2);

      ['list', 'getPokemon'].forEach(prop => {
        expect(PokemonList.propTypes[prop]).toBeTruthy();
      });

    });
  });
});
