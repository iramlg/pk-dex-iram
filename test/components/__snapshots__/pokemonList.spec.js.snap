// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonList /> render test empty list 1`] = `null`;

exports[`<PokemonList /> render test rendering 1`] = `
<div
  className="item-list"
>
  <div
    className="pokemon-item"
    onClick={[Function]}
  >
    <span>
      NAME
    </span>
  </div>
</div>
`;
